blessed/sequences.py

```diff
@@ -339,36 +422,47 @@ def init_sequence_patterns(term):
             # recognize blue_on_red, even if it didn't cause it to be
             # generated.  These are final "ok, i will match this, anyway" for
             # basic SGR sequences.
-            re.escape(u'\x1b') + r'\[(\d+)\;(\d+)\;(\d+)\;(\d+)m',
-            re.escape(u'\x1b') + r'\[(\d+)\;(\d+)\;(\d+)m',
-            re.escape(u'\x1b') + r'\[(\d+)\;(\d+)m',
-            re.escape(u'\x1b') + r'\[(\d+)?m',
-            re.escape(u'\x1b(B'),
+            ('sgr4', re.escape(u'\x1b') + r'\[(\d+)\;(\d+)\;(\d+)\;(\d+)m'),
+            ('sgr3', re.escape(u'\x1b') + r'\[(\d+)\;(\d+)\;(\d+)m'),
+            ('sgr2', re.escape(u'\x1b') + r'\[(\d+)\;(\d+)m'),
+            ('sgr1', re.escape(u'\x1b') + r'\[(\d+)?m'),
+            ('sgr0', re.escape(u'\x1b(B')),
```

Comment:
``s0ds``: Shift to codeset 0 (EUC set 0, ASCII)

http://invisible-island.net/ncurses/terminfo.ti.html#tic-ansi-m
http://invisible-island.net/ncurses/terminfo.ti.html#tic-vt100

```
# This file assumes a US-ASCII character set. If you need to fix this, start
# by global-replacing \E(B and \E)B with the appropriate ISO 6429 enablers
# for your character set.  \E(A and \E)A enables the British character set
# with the pound sign at position 2/3.
```

http://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h2-Controls-beginning-with-ESC

```
C = B  -> United States (USASCII).
```